airbyte-webapp/src/packages/cloud/services/thirdParty/launchdarkly/contexts.ts

```diff
@@ -0,0 +1,46 @@
+import { LDMultiKindContext, LDSingleKindContext } from "launchdarkly-js-client-sdk";
+
+import { User } from "packages/cloud/lib/domain/users/types";
+
+export function createUserContext(user: User | null, locale: string): LDSingleKindContext {
+  const kind = "user";
+
+  if (!user) {
+    return {
+      kind,
+      anonymous: true,
+      locale,
+      // @ts-expect-error LaunchDarkly's typescript types specify that key must be defined, but actually they handle a null or undefined key. It's even recommended in their docs to omit the key for anonymous users https://docs.launchdarkly.com/sdk/client-side/javascript/migration-2-to-3#understanding-changes-to-anonymous-users
+      key: undefined,
```

Comment:
Unless I'm misreading this, this is an error in the LD SDK. It's written in JS, so it's possible this is an oversight by them, types are here: https://github.com/launchdarkly/js-sdk-common/blob/main/typings.d.ts#L349